contracts/contracts/strategies/NativeStaking/CompoundingValidatorManager.sol

```diff
@@ -514,26 +539,78 @@ abstract contract CompoundingValidatorManager is Governable {
-
-        // Calculate the next block timestamp from the verification slot.
-        uint64 nextBlockTimestamp = 12 *
-            verificationSlot +
-            BEACON_GENESIS_TIMESTAMP +
-            // Add 12 seconds for the next block
-            12;
-        // Get the parent beacon block root of the next block which is the block root of the verification slot.
+        require(
+            deposit.slot < depositVerificationSlot,
+            "Slot not after deposit"
+        );
+        require(
+            depositVerificationSlot <= validatorVerificationSlot,
+            "Invalid verification slots"
+        );
+
+        // Get the parent beacon block root of the next block which is the block root of the deposit verification slot.
         // This will revert if the slot after the verification slot was missed.
-        bytes32 blockRoot = BeaconRoots.parentBlockRoot(nextBlockTimestamp);
+        bytes32 depositBlockRoot = BeaconRoots.parentBlockRoot(
+            _calcNextBlockTimestamp(depositVerificationSlot)
+        );
 
         // Verify the slot of the first pending deposit matches the beacon chain
         bool isDepositQueueEmpty = IBeaconProofs(BEACON_PROOFS)
-            .verifyFirstPendingDepositSlot(
-                blockRoot,
-                firstPendingDepositSlot,
-                firstPendingDepositSlotProof
+            .verifyFirstPendingDeposit(
+                depositBlockRoot,
+                firstPendingDeposit.slot,
+                firstPendingDeposit.pubKeyHash,
+                firstPendingDeposit.pubKeyProof
+            );
+
+        // If the deposit queue is not empty
+        if (!isDepositQueueEmpty) {
+            // Get the parent beacon block root of the next block which is
+            // the block root of the validator verification slot.
+            // This will revert if the slot after the verification slot was missed.
+            bytes32 validatorBlockRoot = BeaconRoots.parentBlockRoot(
+                _calcNextBlockTimestamp(validatorVerificationSlot)
             );
 
+            // Verify the validator of the first pending deposit is not exiting.
+            // If it is exiting we can't be sure this deposit has not been postponed in the deposit queue.
+            // Hence we can not verify if the strategy's deposit has been processed or not.
+            IBeaconProofs(BEACON_PROOFS).verifyValidatorWithdrawable(
+                validatorBlockRoot,
+                firstPendingDeposit.validatorIndex,
+                firstPendingDeposit.pubKeyHash,
+                FAR_FUTURE_EPOCH,
+                firstPendingDeposit.validatorWithdrawableEpochProof,
+                firstPendingDeposit.pubKeyProof
+            );
+        }
+
+        // Verify the withdrawableEpoch on the validator of the strategy's deposit
+        IBeaconProofs(BEACON_PROOFS).verifyValidatorWithdrawable(
+            depositBlockRoot,
+            validatorData.index,
```

Comment:
if we verify Index here do we have a need for a separate `verifyValidator` function?

Reply:
This function could promote a validator from STAKED to VERIFIED and add an entry to `verifiedValidators`